src/key_info_managers/sqlite_manager/mod.rs

```diff
@@ -0,0 +1,597 @@
+fn i64_to_auth_type(auth_type: i64) -> Result<AuthType, String> {
+    match FromPrimitive::from_i64(auth_type) {
+        Some(auth_type) => Ok(auth_type),
+        None => Err(format!(
+            "Failed to get AuthType from authenticator_id.\nAuthenticator \"{}\" does not exist.",
+            auth_type
+        )),
+    }
+}
+
+/// SQLite-based `KeyInfoManager`
+///
+/// The `SQLiteKeyInfoManager` relies on access control mechanisms provided by the OS for
+/// the filesystem to ensure security of the database.
+impl SQLiteKeyInfoManager {
+    /// Creates an instance of the sqlite key info manager.
+    /// The SQLiteKeyInfoManager stores key info in the provided database_path file.
+    /// Uses rusqlite.
+    fn new(database_path: PathBuf) -> Result<SQLiteKeyInfoManager> {
+        // Create directory if it does not already exist
+        let mut directory_path = database_path.clone();
+        let _ = directory_path.pop();
+        fs::create_dir_all(&directory_path)?;
+        // Connect to or create database at set path
+        let conn = Connection::open(&database_path)?;
+        let mut key_store = HashMap::new();
+
+        // TODO: Implement kim_metadata table creation here using CURRENT_SCHEMA_VERSION value if
+        // key_mapping & kim_metadata tables do not exist.
+
+        // Create table key_mapping table if it does not already exist
+        let _ = conn.execute(
+            "
+            CREATE TABLE IF NOT EXISTS key_mapping (
+                authenticator_id      INTEGER NOT NULL,
+                application_name      TEXT NOT NULL,
+                key_name              TEXT NOT NULL,
+                provider_uuid         TEXT NOT NULL,
+                provider_name         TEXT NOT NULL,
+                key_info              BLOB NOT NULL,
+                key_info_version      INTEGER NOT NULL,
+                PRIMARY KEY (authenticator_id, application_name, key_name)
+            )
+            ",
+            [],
+        )?;
+
+        let mut stmt = conn.prepare("SELECT * FROM key_mapping")?;
+        let key_iter = stmt.query_map([], |row| {
+            let key_identity = KeyIdentity::new(
+                ApplicationIdentity::new(
+                    row.get("application_name")?,
+                    i64_to_auth_type(row.get("authenticator_id")?).map_err(|e| {
+                        format_error!("Failed to get AuthType from authenticator_id.", e);
+                        let error = Box::new(Error::new(ErrorKind::InvalidData, e));
+                        RusqliteError::FromSqlConversionFailure(64, Integer, error)
+                    })?,
+                ),
+                ProviderIdentity::new(row.get("provider_uuid")?, row.get("provider_name")?),
+                row.get("key_name")?,
+            );
+            let key_info_blob: Vec<u8> = row.get("key_info")?;
+
+            // TODO: Change this to (protobuf?) version once format has been decided.
```

Comment:
Would probably be under the TPM discussions here? https://github.com/parallaxsecond/parsec/issues/504

Reply:
Hmm, I think it should a separate issue that depends on #504 , as they can be done independently.

Reply:
I'll create a new issue, something along the lines of `Decide upon a new serialization format for KeyInfo` sound good?

Reply:
I think the question should actually be - are you planning to implement that in this PR? Or to leave it for a subsequent one? If you're just going to wait to implement it here then there's no need for a new issue. Otherwise something like `Decide and implement...` works

Reply:
Planning on leaving it to a subsequent PR. I'll create an issue.